### PR TITLE
Patch/database functionality fix

### PR DESCRIPTION
**Pull request by toneo-**
This patch fixes some exceptions which were missed during integration, performs some code cleanup and  also takes care of an edge case we missed.
- Bug: SQLService working correctly in some cases and not others. Mostly casing issues with getWhere(). Use upper case table names and lower case column names.
- Bug: UserRepo, MembersRepo, ClaimsRepo not inserting unique IDs causing strange exceptions. There's a new method in SQLService for this
- Bug: The admin could make claims and payments, crashing the web app in the latter case. Now AuthorisationFilter checks if the user is a member and stores this in the session so that member.jsp can prevent non-members from trying to do these things
- Refactor: MembershipPayment takes a combined date/time. The two are still stored separately in the database.

### CODE REVIEW

File: XYZDrivers/src/main/java/com/xyzdrivers/repositories/ClaimsRepo.java [hunk 1]
```diff
@@ -78,16 +78,26 @@ public void insert(Claim c) throws RepositoryException {
...
-            splitClaim[1] = new Date(c.getDate().getTimeInMillis());
-            splitClaim[2] = c.getReason();
-            splitClaim[3] = c.getStatus();
-            splitClaim[4] = c.getAmount();
             
-            String columns = "(MEM_ID, DATE, RATIONALE, STATUS, AMOUNT)";
-        
-
+            // Generate a unique ID from existing row count
+            int nextFreePrimaryKey = sqlService.getNextFreeUniqueInteger("CLAIMS", "id");
```
Comment (lenfred123) [inline XYZDrivers/src/main/java/com/xyzdrivers/repositories/ClaimsRepo.java:83]: I don't understand the need for this - the table id automatically increases. If we're going to do this then we might as well remove the ability for tables to auto increment. And develop does not throw an exception when a claim is submitted to the best of my knowledge.

Comment (bkneis) [reply XYZDrivers/src/main/java/com/xyzdrivers/repositories/ClaimsRepo.java]: Agreed, you should never manually set an id, especially when auto incrementing

File: XYZDrivers/src/main/java/com/xyzdrivers/repositories/PaymentsRepo.java [hunk 2]
```diff
@@ -18,38 +18,61 @@
...
+
+    private List<MembershipPayment> translateMembershipPayments(List<Object[]> flattenedPayments) throws RepositoryException {
+        List<MembershipPayment> payments = new ArrayList<>();
+
+        for (Object[] memberData : flattenedPayments) {
+            Calendar dateTime = combineDateTime(memberData[3], memberData[4]);
+
+            MembershipPayment payment = new MembershipPayment(memberData[0].toString(), // mem_id
+                    memberData[1].toString(), // Type of payment
+                    (float) memberData[2],
```
Comment (bkneis) [inline XYZDrivers/src/main/java/com/xyzdrivers/repositories/PaymentsRepo.java:44]: I think when I was fixing some bugs I saw you can't cast to a float like this even though netbeans doesnt complain

**Review by bkneis — CHANGES_REQUESTED** [review]
LGTM other than the cast and id

File: XYZDrivers/src/main/java/com/xyzdrivers/repositories/PaymentsRepo.java (hunk 2 above)
Comment (toneo-) [reply XYZDrivers/src/main/java/com/xyzdrivers/repositories/PaymentsRepo.java:44]: :(

**Comment by toneo-** [comment]
Closing PR and cherry picking

**Pull request closed without merging**